pkg/apis/operator/v1alpha1/tektonpipeline_lifecycle.go

```diff
@@ -60,6 +76,85 @@ func (tps *TektonPipelineStatus) MarkInstallSucceeded() {
 	}
 }
 
+func (tps *TektonPipelineStatus) MarkCRDsInstalled() {
+	pipelineCondSet.Manage(tps).MarkTrue(CrdInstalled)
+}
+
+func (tps *TektonPipelineStatus) MarkClustersScopedResourcesInstalled() {
+	pipelineCondSet.Manage(tps).MarkTrue(ClustersScoped)
+}
+
+func (tps *TektonPipelineStatus) MarkNamespaceScopedResourcesInstalled() {
+	pipelineCondSet.Manage(tps).MarkTrue(NamespaceScoped)
+}
+
+func (tps *TektonPipelineStatus) MarkDeploymentsInstalled() {
+	pipelineCondSet.Manage(tps).MarkTrue(DeploymentsInstalled)
+}
+
+func (tps *TektonPipelineStatus) MarkWebhookReady() {
+	pipelineCondSet.Manage(tps).MarkTrue(WebhookReady)
+}
+
+func (tps *TektonPipelineStatus) MarkControllerReady() {
+	pipelineCondSet.Manage(tps).MarkTrue(ControllerReady)
+}
+
+func (tps *TektonPipelineStatus) MarkNotReady(msg string) {
+	pipelineCondSet.Manage(tps).MarkFalse(
+		apis.ConditionReady,
+		"Error",
+		"Ready: %s", msg)
+}
+
+func (tps *TektonPipelineStatus) MarkCRDsInstallationFailed(msg string) {
+	tps.MarkNotReady("CRDs installation failed")
+	pipelineCondSet.Manage(tps).MarkFalse(
+		CrdInstalled,
+		"Error",
+		"Install failed with message: %s", msg)
+}
+
+func (tps *TektonPipelineStatus) MarkClustersScopedInstallationFailed(msg string) {
+	tps.MarkNotReady("Cluster Scoped resources installation failed")
+	pipelineCondSet.Manage(tps).MarkFalse(
+		ClustersScoped,
+		"Error",
+		"Install failed with message: %s", msg)
+}
+
+func (tps *TektonPipelineStatus) MarkNamespaceScopedInstallationFailed(msg string) {
+	tps.MarkNotReady("Namespace Scoped resources installation failed")
+	pipelineCondSet.Manage(tps).MarkFalse(
+		NamespaceScoped,
+		"Error",
+		"Install failed with message: %s", msg)
+}
+
+func (tps *TektonPipelineStatus) MarkDeploymentInstallationFailed(msg string) {
+	tps.MarkNotReady("Deployment resources installation failed")
+	pipelineCondSet.Manage(tps).MarkFalse(
+		DeploymentsInstalled,
+		"Error",
+		"Install failed with message: %s", msg)
+}
+
+func (tps *TektonPipelineStatus) MarkWebhookNotAvailable(msg string) {
+	tps.MarkNotReady("Webhook Deployment not available ")
+	pipelineCondSet.Manage(tps).MarkFalse(
+		WebhookReady,
+		"Error",
+		"Webhook: %s", msg)
+}
+
+func (tps *TektonPipelineStatus) MarkControllerNotAvailable(msg string) {
+	tps.MarkNotReady("Controller Deployment not available ")
+	pipelineCondSet.Manage(tps).MarkFalse(
+		ControllerReady,
+		"Error",
+		"Controller: %s", msg)
+}
+
 // MarkInstallFailed marks the InstallationSucceeded status as false with the given
 // message.
 func (tps *TektonPipelineStatus) MarkInstallFailed(msg string) {
```

Comment:
shall we remove functions like these as the respective "coditions" are removed in this pr.

Reply:
some are still there as we user `TektonComponent` so we have to define if we define the interface
but we can do the cleanup as we refactor other components